### ISSUE TITLE
Use Kryo Serialization for FunctionItem and RuntimeIterator

### ISSUE THREAD

**Issue by CanBerker** [RumbleDB/rumble#355]
FunctionItem currently uses Java Serialization for serializing the contained RuntimeIterator.

Additionally, it's not clear whether Spark closures make use of Kryo for the iterators. It would be beneficial for performance to use Kryo in these operations.

The discussions regarding this issue can be found at this PR: https://github.com/RumbleDB/rumble/pull/353

>A place to look at would be the Spark configuration. There are two things that should be configurable: the serialization of results (the Items in the RDDs), which Stefan made use Kryo already during his thesis (plus you did the same for manual serialization in DataFrames), and on the other hand the serialization of all closures passed to Spark in transformations (and this is were the Iterators are serialized), which we are unsure uses Java IO or Kryo.

While I do not have the exact parameters on my mind, this may be a useful reading to start with: 

https://ogirardot.wordpress.com/2015/01/09/changing-sparks-default-java-serialization-to-kryo/

However, since for now the matter with FunctionItems is fixed with using Java IO for the iterator, there is no urgency in this. You can look into this when you see fit.

_Originally posted by @ghislainfourny in https://github.com/RumbleDB/rumble/pull/353_

**Comment by ghislainfourny** [comment]
@CaiJimmy could it be that with your latest changes this issue is now fixed?

**Comment by CaiJimmy** [comment]
> [@CaiJimmy](https://github.com/CaiJimmy) could it be that with your latest changes this issue is now fixed?

Yes, we no longer need to serialise the runtime iterator inside function item to deep copy it.
